engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/SubWorkflowExecutionActor.scala

```diff
@@ -50,18 +50,28 @@ class SubWorkflowExecutionActor(key: SubWorkflowKey,
         subWorkflowStoreActor ! QuerySubWorkflow(parentWorkflow.id, key)
         goto(SubWorkflowCheckingStoreState)
       } else {
-        prepareSubWorkflow(createSubWorkflowId())
+        requestOutputStore(createSubWorkflowId())
       }
   }
 
   when(SubWorkflowCheckingStoreState) {
     case Event(SubWorkflowFound(entry), _) =>
-      prepareSubWorkflow(WorkflowId.fromString(entry.subWorkflowExecutionUuid))
+      requestOutputStore(WorkflowId.fromString(entry.subWorkflowExecutionUuid))
     case Event(_: SubWorkflowNotFound, _) =>
-      prepareSubWorkflow(createSubWorkflowId())
+      requestOutputStore(createSubWorkflowId())
     case Event(SubWorkflowStoreFailure(command, reason), _) =>
       jobLogger.error(reason, s"SubWorkflowStore failure for command $command, starting sub workflow with fresh ID.")
-      prepareSubWorkflow(createSubWorkflowId())
+      requestOutputStore(createSubWorkflowId())
+  }
+  
+  when(WaitingForOutputStore) {
+    case Event(outputStore: OutputStore, SubWorkflowExecutionActorData(Some(subWorkflowId))) =>
+      prepareSubWorkflow(subWorkflowId, outputStore)
+    case Event(outputStore: OutputStore, _) =>
+      jobLogger.error("This is a programmer error, we're ready to prepare the job and should have" +
+        " a SubWorkflowId to use by now but somehow haven't. Starting with a new WorkflowId." +
```

Comment:
It seems like a really sketchy state to be in. OTOH if there's no possible way that it could go awry it'd also be a shame to fail the whole WF if not necessary.